src/components/TapsList.js

```diff
@@ -0,0 +1,120 @@
+// @flow
+
+import type { Tap, TapMutator } from 'brewskey.js-api';
+import type DAOEntityStore from '../stores/DAOEntityStore';
+import type { InfiniteLoaderChildProps } from './InfiniteLoader';
+
+import * as React from 'react';
+import { inject, observer } from 'mobx-react';
+import { withNavigation } from 'react-navigation';
+import { StyleSheet } from 'react-native';
+import { ListItem } from 'react-native-elements';
+import SwipeableFlatList from '../common/SwipeableFlatList';
+import SwipeableActionButton from '../common/SwipeableFlatList/SwipeableActionButton';
+// imported from experimental react-native
+// eslint-disable-next-line
+import SwipeableQuickActions from 'SwipeableQuickActions';
+import InfiniteLoader from './InfiniteLoader';
+
+const styles = StyleSheet.create({
+  // eslint-disable-next-line react-native/no-color-literals
+  listItem: {
+    backgroundColor: 'white',
+  },
+});
+
+type Props = {|
+  tapStore: DAOEntityStore<Tap, TapMutator>,
+  // todo add better typing
+  navigation: Object,
+|};
+
+// todo add pullToRefresh
+@withNavigation
+@inject('tapStore')
+@observer
+class TapsList extends React.Component<Props> {
+  _swipeableFlatListRef: ?SwipeableFlatList;
+
+  _fetchNextData = async (): Promise<void> => {
+    await this.props.tapStore.fetchMany({
+      orderBy: [
+        {
+          column: 'id',
+          direction: 'desc',
+        },
+      ],
+      skip: this.props.tapStore.all.length,
+      take: 20,
+    });
+  };
+
+  _keyExtractor = (item: Tap): string => item.id;
+
+  _onDeleteItemPress = (id: string): Function => async (): Promise<void> => {
+    await this.props.tapStore.deleteByID(id);
+  };
+
+  _onEditItemPress = (id: string): Function => () => {
+    this.props.navigation.navigate('editTap', { id });
+    this._swipeableFlatListRef.resetOpenRow();
+  };
+
+  _onItemPress = (id: string): Function => (): void =>
+    this.props.navigation.navigate('tapDetails', {
+      id,
+    });
+
+  _renderItem = ({ item }: { item: Tap }): React.Node => (
```

Comment:
No sub-render

Reply:
we moving by circle :).
The only way you want it to be a different component is because of onPress handler.
I solve it by (id) =() => close in that handlers and renderItems rerender very few times because of PureSwipableRow component that I made. So, I disagree with you, but `ok`, I'll make those components. :)